tests/unittests/StatsExporterTest.cpp

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "glow/Runtime/StatsExporter.h"
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+using namespace glow;
+
+using testing::ElementsAre;
+
+class MockStatsExporter : public StatsExporter {
+public:
+  MockStatsExporter() { StatsExporter::registerStatsExporter(this); }
+
+  ~MockStatsExporter() override {}
+
+  void addQuantileValue(llvm::StringRef key, double value) override {
+    quantiles[key].push_back(value);
+  }
+
+  void incrementCounter(llvm::StringRef key, int64_t value) override {
+    counters[key] += value;
+  }
+
+  void setCounter(llvm::StringRef key, int64_t value) override {
+    counters[key] = value;
```

Comment:
nit: Making the actual method MOCK_METHOD (but not implementation) gives us better granularity of checking. Like we may do EXPECT_CALL on the method with certain params and actually verify sequence of calls instead of just checking final value of a counter.